### PR TITLE
Add centos7_openstack_support via packer

### PR DESCRIPTION
**Pull request by MalloZup**
Add centos7_image for openstack backend.

# Task:

- [x] packer build should pass

- [x] image should work (deploy boot and sshable on openstack)

- [ ]  do final test with working image with ssh enabled via pwd

### CODE REVIEW

**Comment by MalloZup** [comment]
@Bischoff  if the cloud test went well we are done

**Comment by Bischoff** [comment]
@moio  I just checked, the image for libvirt is indeed of 100 GiB too. Can't we do smaller?

I agree that images should all be of same size in libvirt and openstack. I suppose we even want same images for both, don't we?

**Comment by Bischoff** [comment]
WRT to the SSH problem:

```
Error applying plan:

1 error(s) occurred:

* module.mincentos7.module.minion.null_resource.host_salt_configuration: 1 error(s) occurred:

* ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
```

In `var/log/secure`, polkitd complains of an "Unregistered Authentication agent".

**Comment by Bischoff** [comment]
You are also missing `salt-minion package on the image. epel-release`should be installed first. This package is not needed for `cloud-init`, the comment is wrong.

**Comment by Bischoff** [comment]
There is a problem with packer. It times out at waiting for SSH.

**Comment by moio** [comment]
We need 100 GB on libvirt to accommodate for large cases (RHEL reposyncing). We will be able to reduce this once we use cloud-init for libvirt too (because that can resize disks on first boot).

Cloud images should be as small as possible, 4GB being a good base bet.

**Comment by Bischoff** [comment]
Please see https://github.com/moio/sumaform/pull/269 for a version with many fixes (currently unfinished, but already better).

After fixing the problems with SSH, I stiill get
```
    centos7:   File "/usr/bin/yum", line 28, in <module>
    centos7:     import yummain
    centos7: ImportError: No module named yummain
```

@moio Using 10 GB was a way to get alignement with libvirt images which are currently all 10 GiB. If we want 4 GiB, then we shoud use it everywhere.

**Comment by mcalmer** [comment]
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1       197G   12G  176G   7% /
```
We already use 12G. So maybe 10G is just not enough.

**Comment by Bischoff** [comment]
@mcalmer confusion, stop. I was not speaking about the server.

**Comment by moio** [comment]
> This is just aligning openstack with what we already have in libvirt.

Let me clarify.

In libvirt the image size is defined only in Kiwi/Packer, and it is always 200GB. See:

https://build.suse.de/project/show/Devel:Galaxy:Terraform:Images
https://build.opensuse.org/project/show/systemsmanagement:sumaform:images:libvirt
https://github.com/moio/sumaform/tree/master/packer

This used to be 100GB, it was increased to 200GB due to an edge case. I have now discovered the increase was not correct in all projects, and I fixed it. Please note this is the base image - it makes no difference whether it is then used as a client or server.

In OpenStack we have two sizes: the image size in Kiwi/Packer, which should be big enough to contain the image itself plus a bit of headroom, and the actual disk size at deployment time. In this case, we use 4GB for the former, and the latter is defined by Terraform (currently 10 GB by default on most modules, we can increase it where needed on a module-by-module basis). The resizing from 4GB to 10GB or whatever is configured via Terraform is done by cloud-init on first boot.

To sum it up: I expect new CentOS images for OpenStack defined with 4GB disks in Packer.

One extra thing: I will review the PR as soon as it works and the WIP mark is removed. Ping me if you need help along the way.

**Comment by Bischoff** [comment]
@MalloZup I'm not sure your last commit is useful.

What you need is to add `sshd=1` to packer:
```
      "boot_command": [
        "<tab> sshd=1 text ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ks_centos7.cfg<enter><wait>"
      ]
```

**Comment by MalloZup** [comment]
@Bischoff  why we need that? 

We need btw more space on disk, i have an error

**Comment by MalloZup** [comment]
bildschirmfoto vom 32018-02-12 11-33-04

**Comment by MalloZup** [comment]
@moio  ok this works fine, salt was fine. i pushed latest change with ssh

**Comment by MalloZup** [comment]
this PR can be merged when pass the review, from my pov this is done and we can do another one for pushing the path for downlad image and update the cenots7 variabel

**Comment by moio** [comment]
Fixed and merged, thanks for your help here!
